utils/resolve_file_dependencies.ts

```diff
@@ -0,0 +1,66 @@
+// Copyright 2022 Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {existsSync} from 'fs';
+import {mkdir, readFile, writeFile} from 'fs/promises';
+import {dirname, join, normalize, resolve} from 'path';
+
+const CACHE_DIRECTORY = join(__dirname, '..', '.cache', 'puppeteer');
+
+const args = process.argv.slice(2);
+
+(async () => {
```

Comment:
Please add a comment on the top explaining what this does and what are the input params.